src/score_hv/harvesters/daily_bfg.py

```diff
@@ -155,46 +155,97 @@ def get_data(self):
-            else:
-                units = "None"
-            
-            expected_value, sumweights = np.ma.average(temporal_means,
+            """ The first nested loop iterates through each requested variable.
+               """
+
+            if variable == 'netrf_avetoa': 
+                 """The variable name netrf_avetoa referes to the 
+                    top of the atmosphere net radiative flux.  In this
+                    case we use the var_stats class since we will be
+                    calculating the statistics of the required variables.
+                    """
+                 var_stats_instance = var_stats()
+                 required_vars  = ['dswrf_avetoa','ulwrf_avetoa','uswrf_avetoa'] 
+                 num_vars       = len(required_vars)
+                 for index in range(0,num_vars): 
+                     var_name   = required_vars[index]
+                     var_data   = xr_dataset[var_name]
+                     var_stats_instance.calculate_var_means(var_data,gridcell_area_data)
+                 """The expected value is the weighted averages that are returned from
+                    the function get_weighted_averages in stats_utils..  
+                    It is calculated as follows:
+                    netrf_avetoa = dswrf_avetoa - ulwrf_avetoa - uswrf_avetoa
+                    weighted_averages[0] = weighted averages of dswrf_avetoa
+                    weighted_averages[1] = weighted averages of ulwrf_avetoa
+                    weighted_averages[2] = weighted_averages of uswrf_avetoa
+                 """
+                 weighted_averages = var_stats_instance.get_weighted_averages()
+                 expected_value = weighted_averages[0] - weighted_averages[1] - weighted_averages[2] 
+                 longname           = "toa_net_radiative_flux"
+                 units              = "kg/m**2"
+            else:     
+                 variable_data  = xr_dataset[variable]
+                 var_mean      = np.ma.masked_invalid(variable_data.mean(dim='time',skipna=True))
+                 temporal_means= var_mean                                             
+                 if 'long_name' in variable_data.attrs:
+                     longname = variable_data.attrs['long_name']
+                 else:
+                     longname = "None"
+                 if 'units' in variable_data.attrs:
+                     units = variable_data.attrs['units']
+                 else:
+                      units = "None"
+                 gridcell_area_weights = gridcell_area_data.variables['area']        
+                 expected_value, sumweights = np.ma.average(temporal_means,
                                                   weights=gridcell_area_weights,
                                                   returned=True)
-            assert sumweights >= 0.999 * 4. * np.pi
-            assert sumweights <= 1.001 * 4. * np.pi
+                 assert sumweights >= 0.999 * 4. * np.pi
```

Comment:
why are we using asserts in the main functionality of the code? should the code crash for the user if these values are incorrect? I'm not positive, but I believe best practice would be to not include an assert statement in the main code but instead to use try except clauses with error handling and user friendly error output.